platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalStackDaoImpl.java

```diff
@@ -0,0 +1,232 @@
+        adapters.put(EnvironmentState.class, new EnvironmentStateSerializer());
+        adapters.put(Command.class, new CommandSerializer());
+        adapters.put(Recipe.class, new RecipeSerializer());
+        adapters.put(Limits.class, new LimitsSerializer());
+        adapters.put(MachineSource.class, new MachineSourceSerializer());
+        adapters.put(MachineConfig.class, new MachineSourceSerializer());
+        adapters.put(StackSource.class, new StackSourceSerializer());
+        adapters.put(Permissions.class, new PermissionsSerializer());
+        adapters.put(Group.class, new GroupSerializer());
+
+        this.stackStorage = localStorageFactory.create(STORAGE_FILE,
+                                                       Collections.unmodifiableMap(adapters),
+                                                       Collections.singletonMap(byte[].class, new ByteArrayToBase64Serializer()));
+        this.stacks = new HashMap<>();
+        this.lock = new ReentrantReadWriteLock();
+    }
+
+    @PostConstruct
+    public void start() {
+        stacks.putAll(stackStorage.loadMap(new TypeToken<Map<String, StackImpl>>() {}));
+    }
+
+    @PreDestroy
+    public void stop() throws IOException {
+        stackStorage.store(stacks);
+    }
+
+    @Override
+    public void create(StackImpl stack) throws ConflictException {
+        requireNonNull(stack, "Stack required");
+        lock.writeLock().lock();
+        try {
+            if (stacks.containsKey(stack.getId())) {
+                throw new ConflictException(format("Stack with %s is already exist", stack.getId()));
+            }
+            stacks.put(stack.getId(), stack);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public StackImpl getById(String id) throws NotFoundException {
+        requireNonNull(id, "Stack id required");
+        lock.readLock().lock();
+        try {
+            final StackImpl stack = stacks.get(id);
+            if (stack == null) {
+                throw new NotFoundException(format("Stack with %s was not found", id));
+            }
+            return stack;
+        } finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    @Override
+    public void remove(String id) {
+        requireNonNull(id, "Stack id required");
+        lock.writeLock().lock();
+        try {
+            stacks.remove(id);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public void update(StackImpl update) throws NotFoundException {
+        requireNonNull(update.getId(), "Stack required");
+        lock.writeLock().lock();
+        try {
+            final StackImpl target = stacks.get(update.getId());
+            if (target == null) {
+                throw new NotFoundException(format("Stack with %s was not found", update.getId()));
+            }
+
+            if (update.getName() != null) {
```

Comment:
I think that such type of checks is useless, cause you can set `null` value into `StackImpl` fields.